airflow/contrib/operators/dataproc_operator.py

```diff
@@ -377,6 +377,132 @@ def execute(self, context):
         self._wait_for_done(service)
 
 
+class DataprocClusterScaleOperator(BaseOperator):
+    """
+    Scale a cluster on Google Cloud Dataproc. The operator will wait until the
+    cluster is re-scaled.
+
+    :param cluster_name: The name of the cluster to create.
+    :type cluster_name: string
+    :param project_id: The ID of the google cloud project in which
+        the cluster runs
+    :type project_id: string
+    :param region: leave as 'global', might become relevant in the future
+    :type region: string
+    :param gcp_conn_id: The connection ID to use connecting to Google Cloud Platform.
+    :type gcp_conn_id: string
+    :param num_workers: The new number of workers
+    :type num_workers: int
+    :param num_preepmtible_workers: The new number of preemptible workers
+    :type num_preepmtible_workers: int
+    :param graceful_decommission_timeout: Timeout for graceful YARN decomissioning
+    :type graceful_decommission_timeout: string
+    :param delegate_to: The account to impersonate, if any.
+        For this to work, the service account making the request must have domain-wide
+        delegation enabled.
+    :type delegate_to: string
+    """
+
+    template_fields = ['cluster_name', 'project_id', 'region']
+
+    @apply_defaults
+    def __init__(self,
+                 cluster_name,
+                 project_id,
+                 region='global',
+                 gcp_conn_id='google_cloud_default',
+                 delegate_to=None,
+                 num_workers=2,
+                 num_preemptible_workers=0,
+                 graceful_decommission_timeout=None,
+                 *args,
+                 **kwargs):
+        super(DataprocClusterScaleOperator, self).__init__(*args, **kwargs)
+        self.gcp_conn_id = gcp_conn_id
+        self.delegate_to = delegate_to
+        self.cluster_name = cluster_name
+        self.project_id = project_id
+        self.region = region
+        self.num_workers = num_workers
+        self.num_preemptible_workers = num_preemptible_workers
+        self.graceful_decommission_timeout = graceful_decommission_timeout
+
+    def _wait_for_done(self, service, operation_name):
+        time.sleep(15)
+        while True:
+            response = service.projects().regions().operations().get(
```

Comment:
Can we also have a try.. except blog for 404 errors?

Reply:
This can only happen if the operation is not found, that should never happen, so I think it should raise. Do you that that would be better to add the try..catch block and raise the exception after logging it?

Reply:
Error logged.